### DISCUSSION TITLE
Issues with Relation and applied multiple models

### DISCUSSION THREAD

**Discussion by RfDzDeveloper** [SeaQL/sea-orm#1142] [Q&A] [answered]
Hi! I'm new in Rust, i have many questions. I checked others solutions but don't find my problem(probably?).
I'm lost and i need some help.

Below i uploaded  my repo to give you clear view to my solution.

Firstly I have problem with migration more then two models. When i run sea-orm-cli migrate generate test  to generate migration, i can see only sample migration. Not even full ORM for my model. I already used Django, SQLAlchemy and I think i don't understand how SeaOrm workig(After read the documentation).

Do I need feel migration UP function by my own?

Second part of my question. I checked how to set up Relation between two models. But I'm getting an error:
the trait bound `comment::Entity: Related<post::Entity>` is not satisfied. 
So i put this:
`
#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {
    #[sea_orm(has_many = "super::comment::Entity")]
    Comments,
}`

To my model and still isn't working. 

My models look like below:
`use sea_orm::entity::prelude::*;
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, Eq, DeriveEntityModel, Deserialize, Serialize)]
#[sea_orm(table_name = "Posts")]
pub struct Model {
    #[sea_orm(primary_key)]
    #[serde(skip_deserializing)]
    #[sea_orm(column_name = "PostId")]
    pub post_id: i32,
    #[sea_orm(column_name = "Title")]
    pub title: String,
    #[sea_orm(column_name = "Text")]
    pub text: String,
    #[sea_orm(column_name = "Description")]
    pub description: String,
    #[sea_orm(column_name = "PublishDate")]
    pub publish_date: String,
}

#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {
    #[sea_orm(has_many = "super::comment::Entity")]
    Comments,
}

impl ActiveModelBehavior for ActiveModel {}`

`use sea_orm::entity::prelude::*;
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, Eq, DeriveEntityModel, Deserialize, Serialize)]
#[sea_orm(table_name = "Comments")]
pub struct Model {
    #[sea_orm(primary_key)]
    #[serde(skip_deserializing)]
    #[sea_orm(column_name = "CommentId")]
    pub comment_id: i32,
    #[sea_orm(column_name = "Text")]
    pub text: String,
    #[sea_orm(column_name = "PublishDate")]
    pub publish_date: String,
}

#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {}

impl ActiveModelBehavior for ActiveModel {}
`
I try to do the same thing in different project but i have the same result:
R: EntityTrait + Related,
| ^^^^^^^^^^^^^ required by this bound in has_many

Completely don't understand what should I do with this.

Sorry for laconic questions, but to get an information what exactly to ask, I need know more about language. 
[blog_project-implement_crud.tar.gz](https://github.com/SeaQL/sea-orm/files/9847412/blog_project-implement_crud.tar.gz)

Simply version:
[test_seaorm.tar.gz](https://github.com/SeaQL/sea-orm/files/9854275/test_seaorm.tar.gz)

**Comment by billy1624** [comment, marked as answer]
Hey @RfDzDeveloper, welcome!! You need to update the `comment` entity as well. Please check below.

```rs
use sea_orm::entity::prelude::*;
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, Eq, DeriveEntityModel, Deserialize, Serialize)]
#[sea_orm(table_name = "Comments")]
pub struct Model {
    #[sea_orm(primary_key)]
    #[serde(skip_deserializing)]
    #[sea_orm(column_name = "CommentId")]
    pub comment_id: i32,
    #[sea_orm(column_name = "Text")]
    pub text: String,
    #[sea_orm(column_name = "PublishDate")]
    pub publish_date: String,
    #[sea_orm(column_name = "PostId")]
    pub post_id: i32,
}

#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {
    #[sea_orm(
        belongs_to = "super::post::Entity",
        from = "Column::PostId",
        to = "super::post::Column::PostId"
    )]
    Post,
}

impl Related<super::post::Entity> for Entity {
    fn to() -> RelationDef {
        Relation::Post.def()
    }
}

impl ActiveModelBehavior for ActiveModel {}
```

Docs: https://www.sea-ql.org/SeaORM/docs/relation/one-to-many/

**Reply by RfDzDeveloper** [reply]
Great answer! It was truly helpful :) Big thanks for this!
